tests/queries.py

```diff
@@ -348,3 +348,35 @@ def query_new_iteration(indx, ns):
 
 def query_mem_usage(indx, ns):
     return query_message(indx, ns, ns, {'M': 'json_mem_data'}, False)
+
+
+
+def message_propagation(deployment, fields):
+    logs = query_message(current_index, deployment, deployment, fields, False)
+    srt = sorted(logs,key=lambda x: datetime.strptime(x.T, "%Y-%m-%dT%H:%M:%S.%fZ"))
+    if len(srt) > 0:
+        t1 = datetime.strptime(srt[0].T, "%Y-%m-%dT%H:%M:%S.%fZ")
+        t2 = datetime.strptime(srt[len(srt) - 1].T, "%Y-%m-%dT%H:%M:%S.%fZ")
```

Comment:
This can be written as:
t2 = datetime.strptime(srt[- 1].T, "%Y-%m-%dT%H:%M:%S.%fZ")
Also, better call it with meaningful name: "last_time" (the same for the previous line: "first_time")

Reply:
I didn't really understand

Reply:
In python to get the last element of list: lst[-1]
accordingly, you can access lst[-2] (second element from the end)
etc...